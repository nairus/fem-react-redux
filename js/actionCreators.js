import { SET_SEARCH_TERM, BROWSE_ALL } from './actions';

export function setSearchTerm(searchTerm) {
  return { type: SET_SEARCH_TERM, payload: searchTerm };
}

export function browseAll() {
  return { type: BROWSE_ALL, payload: '' };
}
