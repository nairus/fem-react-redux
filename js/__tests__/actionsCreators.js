// @flow

import { setSearchTerm, addAPIData } from '../actionCreators';

test('setSearchTerm', () => {
  expect(setSearchTerm('New York')).toMatchSnapshot();
});

test('addAPIData', () => {
  expect(
    addAPIData({
      title: 'Game of Thrones',
      year: '2011â€“',
      description: 'Nine noble families fight for control over the mythical lands of Westeros, while a forgotten race returns after being dormant for thousands of years.',
      poster: 'got.jpg',
      imdbID: 'tt0944947',
      trailer: 'giYeaKsXnsI'
    })
  ).toMatchSnapshot();
});
