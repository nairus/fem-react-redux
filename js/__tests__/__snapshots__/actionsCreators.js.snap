// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addAPIData 1`] = `
Object {
  "payload": Object {
    "description": "Nine noble families fight for control over the mythical lands of Westeros, while a forgotten race returns after being dormant for thousands of years.",
    "imdbID": "tt0944947",
    "poster": "got.jpg",
    "title": "Game of Thrones",
    "trailer": "giYeaKsXnsI",
    "year": "2011â€“",
  },
  "type": "ADD_API_DATA",
}
`;

exports[`setSearchTerm 1`] = `
Object {
  "payload": "New York",
  "type": "SET_SEARCH_TERM",
}
`;
